### ISSUE TITLE
CA1416 doesn't support `Debug.Assert(bool, string? message)`

### ISSUE THREAD

**Issue by Bouke** [dotnet/roslyn-analyzers#5403]
I have the following platform guarded code:

```c#
Debug.Assert(OperatingSystem.IsWindows(), "Only supported on windows");
System.ServiceProcess.ServiceBase.Run(service);
```

**Expected behaviour:**  
No diagnostic.

**Actual behaviour:**  
Diagnostic CA1416: "`Run` is only supported on 'windows'"

**Workaround:**  
Remove the message argument:

```c#
Debug.Assert(OperatingSystem.IsWindows());
System.ServiceProcess.ServiceBase.Run(service);
```

However this yields SA1405: "Debug.Assert should provide message text."

**Context:**  
Solution targeting `net5.0`, sdk is at version 5.0.400.

**Analysis:**  
I believe the following line to be relevant. While the method name `IsAnyAssertMethod` suggests that it matches any `Debug.Assert` signature, it only matches the first signature having 1 argument (`Debug.Assert(bool)`).

https://github.com/dotnet/roslyn-analyzers/blob/6b29c673cb8e31865216b64cd3b0a2aca281a32b/src/Utilities/FlowAnalysis/FlowAnalysis/Framework/DataFlow/DataFlowOperationVisitor.cs#L996-L997

https://github.com/dotnet/roslyn-analyzers/blob/6b29c673cb8e31865216b64cd3b0a2aca281a32b/src/Utilities/FlowAnalysis/FlowAnalysis/Framework/DataFlow/DataFlowOperationVisitor.cs#L993-L994

https://github.com/dotnet/roslyn-analyzers/blob/6b29c673cb8e31865216b64cd3b0a2aca281a32b/src/Utilities/FlowAnalysis/FlowAnalysis/Framework/DataFlow/DataFlowOperationVisitor.cs#L262-L263

https://github.com/dotnet/roslyn-analyzers/blob/6b29c673cb8e31865216b64cd3b0a2aca281a32b/src/Utilities/FlowAnalysis/FlowAnalysis/Framework/DataFlow/DataFlowOperationVisitor.cs#L336-L342
